### DISCUSSION TITLE
Samba/CIFS not working (possibly issue with dietpi)

### DISCUSSION THREAD

**Discussion by Paradox** [retronas/discussions#109] [Q&A] [answered]
I am using dietpi on a rpi3b+ I can connect to the default dietpi SMB share, but, the retronas shares won't authenticate, in windows, or, more importantly, MiSTer... This is what I get on MiSTer:

>  Waiting for RETROSMB
> mount: /tmp/cifs_mount: special device //192.168.0.225/mister does not exist.
> //192.168.0.225/mister not mounted
> Done!
> 

Edit: This is the response from cifs_mount.sh, just so there isn't any confusion.

Edit2: dmesg output

> [67185.016257] CIFS: Attempting to mount \\\\192.168.0.225\mister
> [67185.103297] CIFS: VFS:  BAD_NETWORK_NAME: \\\\192.168.0.225\mister
> [67185.125653] CIFS: VFS: cifs_mount failed w/return code = -2

Edit3: I also get this, on the server, when samba starts

> Unknown parameter encountered: "client ntlm auth"
> Ignoring unknown parameter "client ntlm auth"
> Unknown parameter encountered: "ntlmv2 auth"
> Ignoring unknown parameter "ntlmv2 auth"

**Comment by Paradox** [comment]
Before anyone starts answering, I've wiped the sdcard, and, I'm starting over, if I encounter the same problem, I'll come back and confirm, if it works, this time, (I might have done something wrong the first time and borked it) I'll come back and close this discussion.

**Reply by Paradox** [reply]
OK, completely fresh install, no errors, still exactly the same issues.

By the way, in windows, I went into credential manager to remove the stored user/pass for the retrosmb share, and, it seems to still be trying to use it, and doesn't ask for login details, and, also isn't adding a new entry to credential manager, so, I can't even use a different user/pass...

**Comment by birdybro** [comment]
Have you changed the username in the retronas? Password? Hostname? Usually you have to rerun the install scripts and do a few other things to make this happen. Can you run `testparm` from the RetroNAS?

It sounds almost like a DNS hostname resolution issue. Sometimes home routers need to be restarted to resolve this. That might fix the issue. Rebooting the pi, the MiSTer, and your router before trying again could help possibly. Home networks can be finnicky sometimes.

**Reply by birdybro** [reply]
Actually just ran testparm on my home config and on my test debian 11 vm, both have the same error messages, but they work fine. It is likely that the issue is at the DNS side in your network.

**Reply by sairuk** [reply]
I am not a samba dude but perhaps you know @birdybro, is it normal for this output to not include the second slash at the start of the CIFS address? From what I do know, I'd doubt it. DNS is unlikely due to this directly using the IP address

`[67185.103297] CIFS: VFS: BAD_NETWORK_NAME: \192.168.0.225\mister`


Took a look on RH portal, should have two slashes in that output
`kernel: CIFS VFS: BAD_NETWORK_NAME: \\servername\folder`

I don't see anything obviously wrong in the code side of the MiSTer/RetrNAS deploy, I still need to setup cifs shares for ps2 on my rpi yet tho

**Reply by danmons** [reply]
> Actually just ran testparm on my home config and on my test debian 11 vm, both have the same error messages, but they work fine. 

The testarm outputs about the ntlm config items are fine to ignore. I've got a couple of Win95 tests to run, and then I'll clean those up.

**Reply by birdybro** [reply]
> I am not a samba dude but perhaps you know @birdybro, is it normal for this output to not include the second slash at the start of the CIFS address? From what I do know, I'd doubt it. DNS is unlikely due to this directly using the IP address
> 
> `[67185.103297] CIFS: VFS: BAD_NETWORK_NAME: \192.168.0.225\mister`
> 
> Took a look on RH portal, should have two slashes in that output `kernel: CIFS VFS: BAD_NETWORK_NAME: \\servername\folder`
> 

Yeah, good eye.

@Paradox When you filled in the ipaddress part of the cifs_mount.sh script, did you just input it like this:

```ini
SERVER="192.168.0.225"
SHARE="mister"
USER="pi"
PASSWORD="raspberry"
WAIT_FOR_SERVER="true"
MOUNT_AT_BOOT="true"
```

**Reply by sairuk** [reply]
@birdybro `USER` needs to be `USERNAME` or it will result in `sec=none` instead of `user=,password=` passed as mount opts and a failed mount

**Reply by sairuk** [reply]
ok `USER` is wrong in the doco, and needs to be fixed, see example https://github.com/danmons/retronas/wiki/MiSTer-FPGA

**Comment by danmons** [comment]
Double checking you set the RetroNAS password under the "Global configuration" menu AFTER installing the various SMB shares.

This sets both the Linux and SMB/CIFS passwords.

**Reply by sairuk** [reply]
I think our doco should be updated to recommend use of the `cifs_mount.ini` file option instead of directly editing the script so users options persist over potential script updates

as per
```
=========   USER OPTIONS   =========
#You can edit these user options or make an ini file with the same
#name as the script, i.e. mount_cifs.ini, containing the same options.
```

the correct name to use nowdays is `cifs_mount.ini`

**Reply by sairuk** [reply]
See above https://github.com/danmons/retronas/discussions/54#discussioncomment-2128609 also the use of `USER` in the doco example should be `USERNAME` according to cifs_mount.sh otherwise it will result in errors for the end user

**Reply by sairuk** [reply]
Just for completeness it seems @willoucom already brought up the ini approach here https://github.com/danmons/retronas/issues/4#issuecomment-1026530350 and mentioned the correct ini contents including the correct VAR

**Reply by Paradox** [reply]
You all seem to be mostly focused on the MiSTer setup, ignoring the fact that it also can't be accessed from my windows 10 PC, indicating either a failure in the RetroNAS setup, or, as birdybro pointed out, possibly my network setup, itself. only problem is, I don't have problems with the dietpi smb share, from the same box, using the same samba install, or, any other SMB shares on my network, bringing me back to some kind of issue with the RetroNAS shares, themselves.

**Reply by Paradox** [reply]
testparm output

> root@DietPi:~# testparm
> Load smb config files from /etc/samba/smb.conf
> lpcfg_do_global_parameter: WARNING: The "lanman auth" option is deprecated
> lpcfg_do_global_parameter: WARNING: The "client lanman auth" option is deprecated
> Unknown parameter encountered: "client ntlm auth"
> Ignoring unknown parameter "client ntlm auth"
> Unknown parameter encountered: "ntlmv2 auth"
> Ignoring unknown parameter "ntlmv2 auth"
> lpcfg_do_global_parameter: WARNING: The "client ntlmv2 auth" option is deprecated
> Loaded services file OK.
> Weak crypto is allowed
> Server role: ROLE_STANDALONE
> 
> Press enter to see a dump of your service definitions
> 
> # Global parameters
> [global]
>         allow insecure wide links = Yes
>         lanman auth = Yes
>         log file = /var/log/samba/log.%m
>         logging = file
>         map to guest = Bad User
>         max log size = 1000
>         netbios name = RETROSMB
>         ntlm auth = ntlmv1-permitted
>         obey pam restrictions = Yes
>         pam password change = Yes
>         panic action = /usr/share/samba/panic-action %d
>         passwd chat = *Enter\snew\s*\spassword:* %n\n *Retype\snew\s*\spassword:* %n\n *password\supdated\ssuccessfully* .
>         passwd program = /usr/bin/passwd %u
>         server min protocol = CORE
>         server role = standalone server
>         unix extensions = No
>         unix password sync = Yes
>         usershare allow guests = Yes
>         idmap config * : backend = tdb
> 
> 
> [homes]
>         browseable = No
>         comment = Home Directories
>         create mask = 0700
>         directory mask = 0700
>         valid users = %S
> 
> 
> [printers]
>         browseable = No
>         comment = All Printers
>         create mask = 0700
>         path = /var/spool/samba
>         printable = Yes
> 
> 
> [print$]
>         comment = Printer Drivers
>         path = /var/lib/samba/printers
> 
> 
> [retronas]
>         comment = retronas
>         create mask = 0775
>         directory mask = 0775
>         force group = dietpi
>         force user = dietpi
>         include = /etc/samba/retronas.conf
>         path = /data/retronas
>         read only = No
>         wide links = Yes
>         write list = dietpi
> 
> 
> [mister]
>         comment = mister
>         create mask = 0775
>         directory mask = 0775
>         include = /etc/samba/retronas_mister.conf
>         path = /data/retronas/mister
>         read only = No
>         valid users = dietpi
>         wide links = Yes
>         write list = dietpi
>

**Reply by Paradox** [reply]
I changed cifs_mount.ini to mount the dietpi share instead of mister, and it mounted successfully, and I could see the retronas install script in the folder, so, there is something wrong with the mister and retronas shares, because, neither of them will mount, it is not accepting the user/pass what would be the easiest way to remove user/pass and just mount them as guest? I don't need the mister share to be secure.

**Comment by sairuk** [comment]
> You all seem to be mostly focused on the MiSTer setup

Your primary example including error messages is for the mister share, therefore it is reasonable we start there with our attempts to help you. In doing so we have identified errors in the MiSTer documentation that needs correction as such it was captured here.

We do however note that dietpi is untested and unsupported currently so any attempt on our part is to be considered best effort based on the information provided until such time as we move to support other distros. 

I am happy for this particular Discussion to be renamed to "DietPi support discussion" so we can capture what doesnt work quite right in the mean time

The specifics to your issue identified that errors in the dmesg output are as per below, the lack of the second backslash implies an improper setup in the attempt to mount

```
[67185.016257] CIFS: Attempting to mount \192.168.0.225\mister
[67185.103297] CIFS: VFS: BAD_NETWORK_NAME: \192.168.0.225\miste
```

for example if using a mount path with backslashes instead of forward slashes as required from the cli the shell will escape the second slash dropping the first which is what we are seeing here.

so if you are attempting to mount with the following `mount -t cifs -o username=pi,password=****** \\192.x.x.x\mister` from linux or wherever that is incorrect and the end result in the above output will show a single backslash, the correct mounting methos is to use forward slashes '//192.x.x.x/mister` in the mount path

I would suggest as per this question https://github.com/danmons/retronas/discussions/54#discussioncomment-2128329 you go into the Global Configuration and rerun the change password option _after_ installing samba services set the same password you expect to use when prompted and advise results

I will roll a dietpi sdcard this week and see if I can recreate your issue.

**Reply by Paradox** [reply]
I don't know why the first \ is removed in my post, but, the actual dmesg output has both of them.

Edit: looks like a bug in the quote system used by this board.

**Reply by Paradox** [reply]
and, yes, I have done all of the things suggested here. I'm beginning to wonder if there is a bug in the retronas code somewhere where it is trying to use a hardcoded pi in the user/pass somehwre, is there anyone else using a user/pass that isn't pi? I'm going to try creating a pi user, and see if it magically fixes it.

**Reply by sairuk** [reply]
it may, this is alpha software after all so there are expected to be issues, if you haven't run the rename user there may be a situation in one of the scripts where it populates the defaults vs the obtained

**Reply by Paradox** [reply]
I RAN THE DAMN GLOBAL CONFIG

I also created  user/pass pi/pi didn't change a thing, and, yes, I ran all of the scripts again

**Reply by Paradox** [reply]
I guess next I will reimage the card, using the normal pi image and reinstall everything, again, and see if I get the same thing, maybe there is a problem with samba in the repos used by dietpi, although, they should be exactly the same as the repos used by regular pi.

**Reply by sairuk** [reply]
> I RAN THE DAMN GLOBAL CONFIG

I would suggest you do not abuse those 3rd parties taking the time to help you with alpha software on an unsupported distro in the future.

If you find a solution to this I am sure the rest of the community would benefit from it and would appreciate you updating this thread

**Reply by Paradox** [reply]
rpi lite 64bit installs fine, but, seems to shut down after a few minutes, and, i can't figure out why, rpi lite 32 bit complains that my 2GB card isn't 2GB ??? so, I don't know how i'm supposed to test using the official rpi OS.

**Comment by sairuk** [comment, marked as answer]
There are a couple of issues present with dietpi and our current samba config which you are _probably_ running into (based on my dietpi deploy here).

**on the dietpi side** there is no widelinks module installed as a dependency with samba so the setting `wide links = yes` in our samba configs is breaking for dietpi, We have `wide links` enabled for the dos users and EtherDFS

`Error loading module '/usr/lib/aarch64-linux-gnu/samba/vfs/widelinks.so': /usr/lib/aarch64-linux-gnu/samba/vfs/widelinks.so: cannot open shared object file: No such file or directory`

you can work around this by installing the `samba-vfs-modules` package manually we will add this to the samba requirements moving forward.

**on the RetroNAS side** there is a bug where RetroNAS is not updating the pi user in the ansible_vars file and subsequently is failing to set perms on top level file share, this is a problem we will look into separately

**Reply by sairuk** [reply]
https://github.com/danmons/retronas/blob/d05aa6301f0aed9629a319169f69c6d1475439f9/ansible/install_samba.yml#L18 adds the required package for `wide link` support explicilty for OS that does not have the appropriate package dependencies.

**Reply by sairuk** [reply]
Commit https://github.com/danmons/retronas/commit/fe8f0a271854bd88c447380a94dbe4c06382735a fixes the user update issue

**Reply by Paradox** [reply]
That definitely fixed it.
